### PR TITLE
Add link to Databricks type support in Delta Lake connector docs

### PR DESCRIPTION
**Pull request by jhlodin**
## Description

The list of supported types in the Delta Lake connector is directly affected by what's supported in the Delta protocol. This PR links out to the Delta Transaction Log protocol's specification for more information about data type support.

## Non-technical explanation

Provide link out to the Delta protocol documentation for more information about supported types in Delta Lake.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 1]
```diff
@@ -294,6 +294,10 @@ writing data. Data types may not map the same way in both directions between
 Trino and the data source. Refer to the following sections for type mapping in
 each direction.
 
+See the `Databricks data types documentation
+<https://docs.databricks.com/spark/latest/spark-sql/language-manual/sql-ref-datatypes.html>`_
+for more information about supported types.
```
Comment (findepi) [inline docs/src/main/sphinx/connector/delta-lake.rst:299]: The databaricks docs won't know which types are supported by Trino

Comment (mosabua) [reply docs/src/main/sphinx/connector/delta-lake.rst:299]: about available data types in Delta Lake.

Comment (findepi) [reply docs/src/main/sphinx/connector/delta-lake.rst:299]: the paragraphs says "supported" and doesn't indicate it's not about types supported by something else than Trino

Comment (jhlodin) [reply docs/src/main/sphinx/connector/delta-lake.rst:299]: Would "..about types supported by the Delta Lake connector" be specific enough?

Comment (alexjo2144) [reply docs/src/main/sphinx/connector/delta-lake.rst:299]: If I can nit-pick a little more, could be even more clear that we're talking about the Delta Lake format and not the connector:

"for more information about supported data types in the Delta Lake table format specification."

Either way, this is not the right link. I think you want: https://github.com/delta-io/delta/blob/master/PROTOCOL.md#primitive-types

The Databricks link includes all types that are in the SparkSQL language but not all of them are supported in DL

Comment (jhlodin) [reply docs/src/main/sphinx/connector/delta-lake.rst:299]: Reworded and updated the link to point to the Delta protocol specification.

**Review by alexjo2144 — APPROVED** [review]

**Comment by findepi** [comment]
CI https://github.com/trinodb/trino/issues/14618

**Comment by findepi** [comment]
CI https://github.com/trinodb/trino/issues/14618 (second time)

**Pull request merged**